File: Exercise/Medium/08. Word Rotation/Code/Word Rotation.py
import sys

def rotate_right(input_str):
    temp = input_str[-1]
    input_str = input_str[:-1]
    input_str = temp + input_str
    return input_str

def main():
    n = int(input())
    word = input().strip()

    for i in range(n):
        if i > 0:
            print(", ", end="")
        print(word, end="")
        word = rotate_right(word)

    print()

if __name__ == "__main__":
    main()

File: Exercise/Medium/08. Word Rotation/Code/Word Rotation.c
#include <stdio.h>
#include <string.h>

void rotateRight(char* str, int n) {
    char temp = str[n-1];
    for (int i = n-1; i > 0; i--) {
        str[i] = str[i-1];
    }
    str[0] = temp;
}

int main() {
    int n;
    char word[100];
    
    scanf("%d", &n);
    
    scanf("%s", word);

    for (int i = 0; i < n; i++) {
        if (i > 0) printf(", ");
        printf("%s", word);
        rotateRight(word, n);
    }
    printf("\n");
    
    return 0;
}
